ecg/jgi.py

```diff
@@ -374,35 +672,41 @@ def _scrape_urls_unsafe(self, path, domain, organism_urls,
         
         metagenome_domains = ['*Microbiome','cell','sps','Metatranscriptome']
         
-        org_jsons = list()
         pbar = tqdm(organism_urls)
 
+        domain_path = os.path.join(path,domain)
+        organism_url_path = os.path.join(path,domain,'organism_url.json')
+        
         if domain in metagenome_domains:
-            missing_enzyme_data = dict()
+
             for organism_url in pbar:
-                pbar.set_description("Scraping %s ..."%(organism_url))
-                taxon_id, org_dict, missing_enzyme_data = self.__scrape_organism_url_from_metagenome_domain(organism_url,assembly_types,missing_enzyme_data)
+                pbar.set_description("Scraping %s ..."%(re.search(r'\d+$', organism_url).group(0)))
+                taxon_id, org_dict = self.__scrape_organism_url_from_metagenome_domain(domain_path,organism_url,assembly_types)
+
+                organism_urls[organism_url] = 0
```

Comment:
Can't the pruning function be avoided entirely if the organism_url keys are just deleted here instead of set to 0? Is there any reason not to do that?

Reply:
I have the pruning function to check if there are existing files present in the directory and to skip them. I set the organism_url keys to 0 in case there is an error in the last one downloaded and you can just reset the json file to 1 for that organism url by manually editing the file. Also, it lets me test and retest batches of urls by just deleting the json files and resetting their keys to 1 without having to do a large query that is prone to timeouts like Bacteria or Metagenomes.